### PR TITLE
Add Geospatial analysis overview documentation

### PR DESCRIPTION
**Pull request by nreese**
Elasticsearch has powerful geospatial capabilities. There is even a top level marketing page for them https://www.elastic.co/geospatial. However, geo documentation is scattered across various locations within Elasticsearch, Kibana, and ML guides. What's missing is a centralized location that provides an overview of all of Elasticsearch's geospatial capabilities in a single location.

This PR resulted from a standing monthly meeting between engineers and SAs. SAs highlighted the lack of a centralized location highlighting geospatial features. This makes it difficult for prospective customers to compare Elasticsearch with other geo data stores like postgis.

This PR creates a centralized geospatial overview linking to all of existing documentation to fill this gap.

### CODE REVIEW

File: docs/reference/geospatial-analysis.asciidoc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Did you know that {es} has geospatial capabilities? https://www.elastic.co/blog/geo-location-and-search[{es} and geo] go way back, to 2010. A lot has happened since then and today {es} provides robust geospatial capabilities with speed, all with a stack that scales automatically. 
+
+Not sure where to get started with {es} and geo? Then, you have come to the right place.
+
+[discrete]
+[[geospatial-mapping]]
+== Geospatial mapping
+
+Elasticsearch supports two types of geo data: <<geo_point>> fields which support lat/lon pairs, and <<geo_shape>> fields, which support points, lines, circles, polygons, multi-polygons, etc. Use <<explicit-mapping,explicit mapping>> to index geo data fields.
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply docs/reference/geospatial-analysis.asciidoc:15]: Avoid latinisms for accessibility purposes. Check also [How to create accessible web content](https://docs.google.com/document/d/1IGuMf16-cwxx8BX1Z9PWsFWsTPxicwTkX_r7UOhjd4E/edit#heading=h.iykpo698d18b) for more guidance.

**Review by alaudazzi — APPROVED** [review]
Left a few minor editing suggestions, otherwise LGTM.

**Comment by nreese** [comment]
@alaudazzi I am not sure what is causing the docs build to fail in this PR. Would you mind pointing me in the correct direction to resolve the build error?

**Comment by alaudazzi** [comment]
@nreese I asked for help in the #es-docs channel

**Comment by abdonpijpelink** [comment]
This is great. Thanks @nreese !

I've pushed a change that should fix the build error. While I was at it, I've also moved this page up in the TOC, so it comes just after Aggegrations. I think that's a better spot. I've also added a link to this page on the landing page.

**Comment by abdonpijpelink** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.7|[](https://github.com/elastic/elasticsearch/pull/94572)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**